### PR TITLE
fix: apply injection point qualifier on BeanProvider's get call

### LINKED ISSUE #7254: BeanProvider's get by qualifier method ignores injection point qualifier
### Expected Behavior

A qualifier specified on BeanProvider's injection point should participate in bean filtering process when get by qualifier method is called.

### Actual Behaviour

It does not.

### Steps To Reproduce

This is the same as #7132, but in reverse direction.
https://github.com/micronaut-projects/micronaut-core/blob/cd6065195aff610c70865c2f9b194070dc01cbb1/inject/src/main/java/io/micronaut/inject/provider/BeanProviderDefinition.java#L82-L84
Qualifier specified on BeanProvider's injection point is just ignored.

I think that `get`'s method behavior should match `find`'s. 

### Environment Information

- Operation System: `Ubuntu 20.04`
- JDK Version: `GraalVM CE 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05)`

### Example Application

_No response_

### Version

3.4.2

### PR DESCRIPTION
**Pull request by Paullo612**
Fixes #7254

### CODE REVIEW

File: inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy [hunk 1]
```diff
@@ -231,30 +231,91 @@ class BeanOneTwo implements BeanNumber { }
...
+        this.providerTwo = providerTwo;
     }
 }
 ''')
-        when:
+        when: 'retrieve test bean'
         def bean = getBean(context, 'test.Test')
 
-        then:
-        bean.provider.isPresent()
```
Comment (dstepanov) [inline inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: Why did you change this? Preferably do not modify existing checks without explanation

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy (hunk 1 above)
Comment (Paullo612) [reply inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: This is the test for #7133. So, I've improved `get` method test a little bit, and added the same test for `find` method. Changes in this PR touches both `get` and `find` code paths.

I can move changes to `find` method tests to separate commit if you prefer, or rollback them completely.

Comment (dstepanov) [reply inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: Just return that `bean.provider.isPresent()` check

Comment (Paullo612) [reply inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: Ok, may I force force push? Or I should do a separate commit?

Comment (dstepanov) [reply inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: You can force push

Comment (Paullo612) [reply inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy:243]: Done. Also added the same check to `get` method test.

**Review by dstepanov — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**